app/models/manageiq/providers/vmware/infra_manager/provision/configuration/disk.rb

```diff
@@ -21,13 +21,38 @@ def build_config_disk_spec(vmcs)
     end
   end
 
-  def get_scsi_controller_info
-    inventory_hash = source.with_provider_connection do |vim|
-      vim.virtualMachineByMor(source.ems_ref_obj)
+  def build_disk_relocate_spec(datastore)
+    VimArray.new('ArrayOfVirtualMachineRelocateSpecDiskLocator') do |relocate_spec_array|
+      (disks_info).map do |disk|
```

Comment:
@d-m-u if you're just using the disks from the source to get the disk.key then we should be able to get that from the database

Reply:
I actually can't believe it but we don't have the device.key stored in the disks table, we really should.

Reply:
Can you open a schema PR to add an ems_ref to disks and guest_devices?  We'll use that to store the disk.key.

Reply:
https://github.com/ManageIQ/manageiq-schema/pull/402 and https://github.com/ManageIQ/manageiq-schema/pull/403

Reply:
:+1: thanks